pkg/frontend/querymiddleware/results_cache.go

```diff
@@ -365,6 +371,8 @@ func mergeCacheExtentsForRequest(ctx context.Context, r MetricsQueryRequest, mer
 
 	for i := 1; i < len(extents); i++ {
 		if accumulator.End+r.GetStep() < extents[i].Start {
+			// The response is marsheld in mergeCacheExtentsWithAccumulator, so we can safely close it afterwards
+			defer accumulator.Close()
```

Comment:
Should we call `accumulator.Close` after the error check below? Otherwise this deferred `Close` won't happen until `mergeCacheExtentsForRequest` returns.

Reply:
Actually this was the wrong place to close. The response is still needed after mergeCacheExtentsForRequest and the merged responses are closed further downstream already so this is removed now.